snakemake/scheduler.py

```diff
@@ -522,11 +523,103 @@ def exit_gracefully(self, *args):
             self._user_kill = "graceful"
         self._open_jobs.release()
 
+    def job_selector_ilp(self, jobs):
+        """
+        Job scheduling by optimization of resource usage by solving ILP using pulp 
+        """
+        import pulp
+        import math
+        from pulp import lpSum
+
+        # assert self.resources["_cores"] > 0
+        scheduled_jobs = [
```

Comment:
If you change this into a dict that has the job object as a key and the LpVariable as a value, you don't have to do the split on the name below, right?

Reply:
Done